gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceExtractor.java

```diff
@@ -581,9 +593,12 @@ public String getTimestampPredicateCondition(String column, long value, String v
 
       // Get data from input stream
       // If bulk load is not finished, get data from the stream
-      if (!this.isBulkJobFinished()) {
-        rs = getBulkData();
-      }
+      // Skip empty result sets since they will cause the extractor to terminate early
```

Comment:
Is it better to do the skip loop in `getBulkData`

Reply:
`getBulkData` has a couple places where it returns, so I thought that this was cleaner and less error prone.